File: web/vagao-web/src/main/java/com/vagao/entidade/Usuario.java
package com.vagao.entidade;

import java.io.Serializable;

/**
 * Entidade que representa um usuário do sistema (cliente ou administrador).
 * Espelha a tabela `usuario` do banco de dados.
 */
public class Usuario implements Serializable {

    private static final long serialVersionUID = 1L;

    private int idUsuario;
    private String nome;
    private String email;
    private String senha;
    private String perfil;
    private String telefone;

    public Usuario() {
    }

    public int getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(int idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

    public String getPerfil() {
        return perfil;
    }

    public void setPerfil(String perfil) {
        this.perfil = perfil;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public boolean isAdmin() {
        return "admin".equals(perfil);
    }

    public boolean isCliente() {
        return "cliente".equals(perfil);
    }
}

File: mobile/vagao_app/lib/modelos/usuario.dart
/// Espelha a entidade com.vagao.entidade.Usuario do backend Java.
/// Nunca carrega senha — a API não a envia (JsonUtil.usuarioParaJson a omite).
class Usuario {
  final int idUsuario;
  final String nome;
  final String email;
  final String perfil;
  final String? telefone;

  Usuario({
    required this.idUsuario,
    required this.nome,
    required this.email,
    required this.perfil,
    this.telefone,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json['idUsuario'] as int,
      nome: json['nome'] as String,
      email: json['email'] as String,
      perfil: json['perfil'] as String,
      telefone: json['telefone'] as String?,
    );
  }

  /// Espelha Usuario.isAdmin() do backend — mesma regra dos dois lados.
  bool get isAdmin => perfil == 'admin';

  /// Espelha Usuario.isCliente() do backend — mesma regra dos dois lados.
  bool get isCliente => perfil == 'cliente';
}
